pkg/operator/controller/ingress/status_test.go

```diff
@@ -2215,51 +2216,95 @@ func Test_checkZoneInConfig(t *testing.T) {
 }
 
 func Test_computeIngressUpgradeableCondition(t *testing.T) {
-	makeDefaultCertificateSecret := func(cn string, sans []string, signatureAlgorithm x509.SignatureAlgorithm) *corev1.Secret {
+	makeCertificateSecret := func(certs ...*x509.Certificate) *corev1.Secret {
+		var certData []byte
+		for _, cert := range certs {
+			pemBlock := &pem.Block{
+				Type:  "CERTIFICATE",
+				Bytes: cert.Raw,
+			}
+			certData = append(certData, pem.EncodeToMemory(pemBlock)...)
+		}
+		return &corev1.Secret{
+			Data: map[string][]byte{
+				"tls.crt": certData,
+			},
+		}
+	}
+	makeCertificate := func(cn string, sans []string, signatureAlgorithm x509.SignatureAlgorithm, rootCert *x509.Certificate, rootCertPrivateKey crypto.PrivateKey, isCA bool) (*x509.Certificate, any) {
 		certTemplate := &x509.Certificate{
 			SerialNumber:       big.NewInt(1),
 			Subject:            pkix.Name{CommonName: cn},
 			DNSNames:           sans,
 			SignatureAlgorithm: signatureAlgorithm,
 		}
-
-		var cert []byte
+		if isCA {
+			certTemplate.IsCA = true
+			certTemplate.KeyUsage = x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign
```

Comment:
I believe you also need `certTemplate.BasicConstraintsValid = true`.

Reply:
Yup, thanks, done.